src/user/phone-validation/phoneNumberTransform.interceptor.ts

```diff
@@ -0,0 +1,25 @@
+import { CallHandler, ExecutionContext, NestInterceptor } from '@nestjs/common';
+import { Request } from 'express';
+import { Observable } from 'rxjs';
+
+export class PhoneNumberTransformInterceptor implements NestInterceptor {
+  intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
+    const request = context.switchToHttp().getRequest();
+    this.transformPhoneNumber(request);
+    return next.handle();
+  }
+
+  private transformPhoneNumber(request: Request): void {
+    const phoneNumber = request.body?.phoneNumber;
+    if (phoneNumber && typeof phoneNumber === 'string') {
+      request.body.phoneNumber = this.formatPhoneNumber(phoneNumber);
+    }
```

Comment:
shouldn't we throw an exception if phoneNumber isn't string? Or maybe it already handled in some validator?

Reply:
yes, it is already handled by a validator, thats why i didnt write it here